### PR TITLE
feat: use FieldLookupBackend from ansible_base to filter eda credetials

### PR DESCRIPTION
**Pull request by hsong-rh**
https://issues.redhat.com/browse/AAP-23933

### CODE REVIEW

**Comment by hsong-rh** [comment]
Wait for PR https://github.com/ansible/django-ansible-base/pull/384

File: src/aap_eda/api/views/eda_credential.py [hunk 1]
```diff
@@ -63,10 +57,16 @@ class EdaCredentialViewSet(
...
     filter_backends = (
-        KindFilterBackend,
+        FieldLookupBackend,
         defaultfilters.DjangoFilterBackend,
     )
     filterset_class = filters.EdaCredentialFilter
+    rest_filters_reserved_names = [
+        "refs",
+        "force",
+        "name",
```
Comment (mkanoor) [inline src/aap_eda/api/views/eda_credential.py:67]: @hsong-rh Why is name a reserved attribute, shouldn't customers be allowed to filter by name?

Comment (hsong-rh) [reply src/aap_eda/api/views/eda_credential.py:67]: no, ANSIBLE_BASE_REST_FILTERS_RESERVED_NAMES don't contain `name` and `credential_type_id`.

**Comment by hsong-rh** [comment]
The work moved to https://github.com/ansible/eda-server/pull/912

**Pull request closed without merging**